lib/HtmlPanel.tsx

```diff
@@ -0,0 +1,94 @@
+import { useContext, useEffect, useState } from "react";
+import { createPortal } from "react-dom";
+import Talk from "talkjs";
+import { BoxContext } from "./MountedBox";
+
+type HtmlPanelProps = {
+  /**
+   * The URL you want to load inside the HTML panel. The URL can be absolute or
+   * relative. We recommend using same origin pages to have better control of
+   * the page. Learn more about HTML Panels and same origin pages {@link https://talkjs.com/docs/Features/Customizations/HTML_Panels/ | here}.
+   */
+  url: string;
+
+  /** The panel height in pixels. Defaults to `100px`. */
+  height?: number;
+
+  /** Sets the visibility of the panel. Defaults to `true`. */
+  show?: boolean;
+
+  /** If given, the panel will only show up for the conversation that has an `id` matching the one given. */
+  conversationId?: string;
+
+  /** The content that gets rendered inside the `<body>` of the panel. */
+  children: React.ReactNode;
+};
+
+type State =
+  | { type: "none" }
+  | { type: "loading" }
+  | { type: "loaded"; panel: Talk.HtmlPanel };
+
+export function HtmlPanel({
+  url,
+  height = 100,
+  show = true,
+  conversationId,
+  children,
+}: HtmlPanelProps) {
+  const [state, setState] = useState<State>({ type: "none" });
+  const box = useContext(BoxContext);
+
+  useEffect(() => {
+    async function run() {
+      if (state.type !== "none" || !box) return;
+
+      setState({ type: "loading" });
+      const panel = await box.createHtmlPanel({
+        url,
+        conversation: conversationId,
+        height,
+        show,
+      });
+      await panel.windowLoadedPromise;
+      setState({ type: "loaded", panel });
+    }
+
+    run();
+
+    return () => {
+      if (state.type === "loaded") {
+        state.panel.destroy();
```

Comment:
shouldn't we always destroy it, regardless of loading state? else if customer code creates it and then quickly removes it, the chatbox will create it anyway.

Reply:
(maybe just put the `Promise<HtmlPanel>` in the state, and do `state.panelPromise.then(panel => panel.destroy())`. i don't think there's a downside to already-resolved panels being destroyed in the next microtask instead of right there in the cleanup callback, right?